### DISCUSSION TITLE
Filtering the request body in resteasy reactive

### DISCUSSION THREAD

**Discussion by ichrist-gr** [quarkusio/quarkus#29828] [Q&A] [answered]
Hello folks,

In my organization we try to implement a request body filtering and logging for all the requests invoked towards our microservices. 
We use **resteasy reactive jackson** and we can not read the entity stream because it is a **blocking** action.

What we have made so far is:
1) Inject the context io.vertx.core.http.HttpServerRequest;
2) Read the request body in a non-blocking way (httpServerRequest.body())

It works smoothly when request bodies are **present** in the request, but when we try to fire a GET method with **no request body**, the Future<Buffer> that vert.x returns from invoking httpServerRequest.body(), **can't be completed**.

So we would like to find a workaround, to filter if the body is empty or not, because if httpServerRequest.body() has no payload, on complete method will never be triggered.

The current state of the code looks like this:
`@Override
    public void filter(ContainerRequestContext containerRequestContext) throws IOException {
        ULFFRecord record = createULFFRecord(containerRequestContext, resourceInfo, conf);

        if (printPayload(record.getHttpMethod())) {
            httpServerRequest.body()
                    .onComplete(event -> {
                        if (event.succeeded()) {
                            record.setPayload(event.result().toString());
                        }
                        vfLogger.logUlffStandard(record);
                    });
        } else {
            vfLogger.logUlffStandard(record);
        }
    }`
`private boolean printPayload(String httpMethod) {
        boolean deleteAction = HttpMethod.DELETE.equals(httpMethod);
        boolean getAction = HttpMethod.GET.equals(httpMethod);

        return !Boolean.logicalOr(getAction, deleteAction) && conf.requestBodyLoggingEnabled();
    }`

Let's include also, that this feature is going to be a common library for all our endpoints, so we need something magic that plays everywhere.

[a comment by an automated account is not shown]

**Comment by ichrist-gr** [comment, marked as answer]
Finally, an answer to this is to retrieve the http header "Content-Length" and check if it's greater than 0

**Reply by stuartwdouglas** [reply]
Can you just call `isEnded()` on the request to check? It's not common, but the request could also have a `Transfer-Encoding: chunked` header instead.
